src/services/WalletService/deterministic/useDeterministicWallet.ts

```diff
@@ -92,10 +91,33 @@ const useDeterministicWallet = (
     }
   }, [state.isInit]);
 
+  // On first connection && on asset update
   useEffect(() => {
```

Comment:
Can we add tests for some of this? Goes for the entire feature. It is complex logic and one of the least tested parts of the app.

Reply:
dun did :3